### PR TITLE
Skip submariner log collection if its not deployed at all

### PR DESCRIPTION
**Pull request by shylesh**

### CODE REVIEW

**Review by petr-balogh — APPROVED** [review]

**Review by vavuthu — DISMISSED** [review]

File: ocs_ci/ocs/utils.py [hunk 1]
```diff
@@ -1152,6 +1152,22 @@ def _collect_ocs_logs(
                 acm_mustgather_path, acm_mustgather_image, cluster_config=cluster_config
             )
 
+        # Check if subctl exists
+        # If we have deployed submariner then we should have binary in the
+        # path else submariner may not be in the picture and we will skip log collection
+        try:
+            run_cmd("submariner -h")
```
[the remark this answers is not in the document]
Comment (shylesh) [reply ocs_ci/ocs/utils.py:1159]: oops, yes, my bad.

[the remark this answers is not in the document]
Comment (shylesh) [reply ocs_ci/ocs/utils.py]: done

File: ocs_ci/ocs/utils.py [hunk 2]
```diff
@@ -1152,6 +1152,22 @@ def _collect_ocs_logs(
...
+        except FileNotFoundError:
+            log.warning(
+                "Submariner binary doesn't exists, checking if submariner pods exists..."
+            )
+            out = run_cmd("oc get pods -n submariner-operator")
+            if "No resources" in out:
```
[the remark this answers is not in the document]
Comment (shylesh) [reply ocs_ci/ocs/utils.py:1166]: done

**Review by petr-balogh — APPROVED** [review]

**Review by petr-balogh — DISMISSED** [review]

**Review by petr-balogh — DISMISSED** [review]

File: ocs_ci/deployment/acm.py [hunk 3]
```diff
@@ -249,3 +249,35 @@ def get_default_gateway_node(self):
         """
         # Always return the first worker node
         return get_typed_worker_nodes()[0]
+
+
+def download_subctl_binary(source="upstream"):
+    if source == "upstream":
```
Comment (petr-balogh) [inline ocs_ci/deployment/acm.py:255]: Maybe we can add else block with raising exception if the other source will be provided as we currently support only the `upsteram`.

File: ocs_ci/ocs/utils.py [hunk 4]
```diff
@@ -1257,6 +1257,29 @@ def _collect_ocs_logs(
...
+                    "Not able to find submariner resources, skipping submariner log collection"
+                )
+                return
+            else:
+                # submariner resources exists but subctl binary missing
+                # download subctl
+                # Importing here to avoid circular imports issue and tox failure
+                from ocs_ci.deployment.acm import download_subctl_binary
+
+                download_subctl_binary()
```
Comment (petr-balogh) [inline ocs_ci/ocs/utils.py:1281]: Get clarity about the US/DS submariner and we need to pass here probably the value from config and also probably need to have the way to get the DS subctl as well - if it's not possible at the moment, we need to as W/A use US subctl only but at some point we should implement download subctl also for DS.

**Pull request closed without merging**